src/Controller/ICalendarController.php

```diff
@@ -0,0 +1,287 @@
+<?php
+
+namespace App\Controller;
+
+use App\Controller\BaseController;
+use App\Model\Agent;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\Routing\Annotation\Route;
+
+require_once(__DIR__.'/../../public/conges/class.conges.php');
+require_once(__DIR__.'/../../public/personnel/class.personnel.php');
+require_once(__DIR__.'/../../public/postes/class.postes.php');
+
+class ICalendarController extends BaseController
+{
+    /**
+     * @Route("ical", name = "ical.index", methods={"GET"})
+     */
+    public function index(Request $request, Session $session){
+
+        if (!$this->config('ICS-Export')) {
+            return $this->returnError("L'exportation ICS est désactivée", 403);
+        }
+
+        $interval_get = $request->get('interval');
+        $code = $request->get('code');
+        $id = $request->get('id');
+        $login = $request->get('login');
+        $mail = $request->get('mail');
+        $get_absences = $request->get('absences');
+
+        $agent = null;
+
+        // Définition de l'id de l'agent si l'argument login est donné
+        if (!$id and $login) {
+            $agent = $this->entityManager->getRepository(Agent::class)->findOneBy(array('login' => $login));
+            if ($agent) {
+                $id = $agent->id();
+            } else {
+                return $this->returnError("Impossible de trouver l'id associé au login $login", 400);
+            }
+        }
+
+        // Définition de l'id de l'agent si l'argument mail est donné
+        if (!$id and $mail) {
+            $agent = $this->entityManager->getRepository(Agent::class)->findOneBy(array('mail' => $mail));
+            if ($agent) {
+                $id = $agent->id();
+            } else {
+                return $this->returnError("Impossible de trouver l'id associé au mail $mail", 400);
+            }
+        }
+
+        if (!$agent && $id) {
+            $agent = $this->entityManager->getRepository(Agent::class)->find($id);
+            if ($agent) {
+                $id = $agent->id();
```

Comment:
L'affectation $id = $agent->id(); me semble inutile.
Nous pouvons simplifier par if(empty($agent)) { return $this->returnError ...}